thread 'rustc' panicked at 'assertion failed: !substs.has_escaping_bound_vars()'
error: internal compiler error: unexpected panic
note: the compiler unexpectedly panicked. this is a bug.